src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -341,6 +345,36 @@ func (mod *mgmtModule) getFabricInterface(ctx context.Context, params *FabricIfa
 	return mod.cache.GetFabricDevice(ctx, params)
 }
 
+func (mod *mgmtModule) populateNUMAFabricMap(ctx context.Context, resp *mgmtpb.GetAttachInfoResp) error {
+	numaMap, unlockMap, err := mod.cache.GetNUMAFabricMap(ctx, hardware.NetDevClass(resp.ClientNetHint.NetDevClass), resp.ClientNetHint.Provider)
+	if err != nil {
+		return err
+	}
+	defer unlockMap()
+
+	resp.NumaFabricInterfaces = make(map[uint32]*mgmtpb.FabricInterfaces)
+	for numaNode, fis := range numaMap {
+		pbFIs := &mgmtpb.FabricInterfaces{
+			Ifaces: make([]*mgmtpb.FabricInterface, 0, len(fis)),
+		}
+
+		for _, fi := range fis {
+			if fi.HasProvider(resp.ClientNetHint.Provider) {
+				pbFIs.Ifaces = append(pbFIs.Ifaces, &mgmtpb.FabricInterface{
+					NumaNode:  uint32(numaNode),
+					Interface: fi.Name,
+					Domain:    fi.Domain,
+					Provider:  resp.ClientNetHint.Provider,
+				})
+			}
+		}
+
+		resp.NumaFabricInterfaces[uint32(numaNode)] = pbFIs
+	}
```

Comment:
This is a good point. We can do this filtering earlier in the game to ensure everything in the cached NUMAFabric struct matches the expectations (dev class and provider combination).

Reply:
In my latest changes, I remove any non-matching net dev class when we cache the map. I decided that we should trust the fabric interface provider to have given us what we asked for, as far as fabric provider filtering.

Do you think we need to go further than this? I.e. maybe cache the fabric and GetAttachInfoResp together in protobuf format? Right now we're caching the `control.GetAttachInfoResp` rather than the protobuf one. It seems to me that it would be okay to cache the protobuf response in toto, but that's a bit more surgery than I wanted to do in this feature addition PR.

Reply:
I don't see any problem with caching response in protobuf format it makes things easier

Reply:
> It seems to me that it would be okay to cache the protobuf response in toto

I tend to agree. This particular control RPC is a bit of an oddball in that there isn't really a strong use case for converting between protobuf and the native response type. That having been said, it would be fine with me to pursue further optimizations in a separate PR.